execution/tests/posix/execution.cc

```diff
@@ -726,3 +726,23 @@ SCENARIO("executing commands with execution::each_line") {
         }
     }
 }
+SCENARIO("Toggle expanding command paths behaviour in execution::expand_command") {
+    GIVEN("set expand to false with a built-in executable") {
+        THEN("the executable is not expanded to an absolute path") {
+            boost::format fmt = boost::format("cd %1% && ls") % EXEC_TESTS_DIRECTORY;
+            std::string fmtStr = boost::str(fmt);
+            vector<string> lines;
+            string command = expand_command(fmtStr, lines, false );
+            REQUIRE( command == "cd " EXEC_TESTS_DIRECTORY " && ls");
+        }
+    }
+    GIVEN("set expand to true with a built-in executable") {
+        THEN("the executable absolute path is not found") {
+            boost::format fmt = boost::format("cd %1% && ls") % EXEC_TESTS_DIRECTORY;
+            std::string fmtStr = boost::str(fmt);
+            vector<string> lines;
+            string command = expand_command(fmtStr, lines, true );
+            REQUIRE( command == "");
+        }
+    }
+}
```

Comment:
there's already a SCENARIO that tests this function, we can move these tests there for better readability:
https://github.com/puppetlabs/leatherman/blob/b1d1ea2bb8789ea602594acc8a3c8b8a8008ee67/execution/tests/posix/execution.cc#L83

Reply:
moved